app/assets/javascripts/oxalis/model/binary/layers/wk_layer.js

```diff
@@ -83,11 +84,17 @@ class WkLayer extends Layer {
   }
 
   async sendToStoreImpl(batch: Array<DataBucket>): Promise<void> {
-    const items = batch.map(bucket => {
+    let counter = 0;
+    const items = [];
+    for (const bucket of batch) {
+      counter++;
+      // Do not block the main thread for too long as Base64.fromByteArray is performance heavy
+      // eslint-disable-next-line no-await-in-loop
+      if (counter % 3 === 0) await Utils.sleep(1);
```

Comment:
Can you make `3` a named constant? 
How did you come up with that? Did you measure how long it takes to encode one bucket?

Reply:
Yes. Encoding three buckets takes ~15ms on average and no lag is noticeable.